pkg/cloudprovider/aws/instance.go

```diff
@@ -36,16 +36,20 @@ type InstanceProvider struct {
 // Create an instance given the constraints.
 func (p *InstanceProvider) Create(ctx context.Context,
 	launchTemplate *ec2.LaunchTemplate,
-	instanceTypeOptions []string,
+	instanceTypeOptions map[string][]*ec2.InstanceTypeInfo,
```

Comment:
consider calling this zonalInstanceTypeOptions for parity with zontalSubnetOptions